#!/usr/bin/env python
PACKAGE = "navigation"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

#PathPreprocessing
gen.add("tracking_distance_polynomial", double_t, 0, "Discard remembered path points after this distance [m]", 10000, 0, 5)
gen.add("tracking_distance_behind", double_t, 0, "Discard remembered gates with distance [m] behind vehicle", 10000, 0, 5)
gen.add("tracking_distance_in_front", double_t, 0, "Discard remembered gates with distance [m] in front of vehicle", 10000, 0, 5)
gen.add("corridor_simplification_max_merge_distance", double_t,       0, "Maximal allowed distance between to close gate centers in m", 10000, 0, 0.1)
gen.add("use_adaptive_polynomial_degree", bool_t,       0, "use adaptive polynomial degree", True)
gen.add("path_polynomial_degree", int_t,       0, "degree for polynomial approximates lane course if degree is not adaptive", 0, 0, 10)
gen.add("enable_perpendicular_parking", bool_t,       0, "plan parking maneuver when parking slot is available", False)
gen.add("enable_turning", bool_t,       0, "plan turning maneuver", False)

exit(gen.generate(PACKAGE, "path_preprocessing", "PathPreprocessing"))
 
